Tests/YPersistenceTests/Manager/PersistenceManager+SyncTests.swift

```diff
@@ -92,4 +92,56 @@ final class PersistenceManagerSyncTests: PersistenceManagerBaseTests {
             XCTAssertTrue(product.wasDeleted)
         }
     }
+
+    func test_markRecordAsUploaded_deliversMarkedForUploadedRecords() throws {
+        try confirmFruitsEmpty()
+        try insertFruits()
+
+        var allProducts: [ManagedFruit] = try sut.fetchRecordsToUpload()
+        XCTAssertEqual(allProducts.count, 4)
+
+        let uids = ["1", "2"]
+        try sut.markRecordsAsUploaded(enity: ManagedFruit.self, uids: uids)
+
+        allProducts = try sut.fetchRecordsToUpload()
+        XCTAssertEqual(allProducts.count, 2) // with uid 1 and 2
+
+        for product in allProducts {
+            XCTAssertFalse(product.isUploaded)
+        }
+    }
+    
+    func test_markRecordAsDelete_deliversMarkedForDeleteRecords() throws {
+        try confirmFruitsEmpty()
+        try insertFruits()
+
+        let uids = ["1", "2", "x"]
+        try sut.markRecordsAsDeleted(enity: ManagedFruit.self, uids: uids)
+
+        let allProducts: [ManagedFruit] = try sut.fetchRecordsToDelete()
+        XCTAssertEqual(allProducts.count, 2)
+
+        for product in allProducts {
+            XCTAssertTrue(product.wasDeleted)
+        }
```

Comment:
```suggestion
```
Not needed.
But let's include fetching before the change and asserting count == 0 (or isEmpty) like in the previous test.